### ISSUE TITLE
Test succeeds when setupSpec throws

### ISSUE THREAD

**Issue by DieterVDW** [spockframework/spock#1209]
# Issue description
When throwing an exception in the `setupSpec` method, the test class is marked as success when running with Maven

Probably an issue with how Maven / Groovy / Junit / Spock are integrated, but I have no clue where to start looking...

Quite annoying to discover a bunch of tests are failing silently...

# How to reproduce
See minimal test case here: https://github.com/DieterVDW/spock-test

## Link to a gist or similar (optional)
TL;DR: The following test class succeeds when building with Maven

```
import spock.lang.Specification

class SpockTest extends Specification {
    def setupSpec() {
      throw new IllegalArgumentException()
    }

    def theTest() {
        when:
        def x = 10

        then:
        x == 10
    }
}
```

```
[INFO] Running SpockTest
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.321 s - in SpockTest
```

# Additional Environment information
```
~ mvn -version
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /usr/local/Cellar/maven/3.6.3_1/libexec
Java version: 13.0.2, vendor: N/A, runtime: /usr/local/Cellar/openjdk/13.0.2+8_2/libexec/openjdk.jdk/Contents/Home
Default locale: en_SE, platform encoding: UTF-8
OS name: "mac os x", version: "10.15.6", arch: "x86_64", family: "mac"
```

## Groovy version
3.0.3

## Build tool version
Maven 3.6.3

## Operating System
OSX Catalina 10.15.6

## IDE
N/A

### Apache Maven
            <dependency>
                <groupId>org.spockframework</groupId>
                <artifactId>spock-bom</artifactId>
                <version>${spock.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>

**Comment by leonard84** [comment]
From a first test with Intellij it gets reported correctly. It seems to be a problem with maven-surefire.
@marcphilipp do you know about such problems with surefire?

**Comment by marcphilipp** [comment]
> do you know about such problems with surefire?

Yes, that should be fixed in 3.0.0-M5: https://issues.apache.org/jira/browse/SUREFIRE-1741

**Comment by DieterVDW** [comment]
@marcphilipp : If only all problems were this easy to solve!
Bumping the version to 3.0.0-M5 fixes the issue!
